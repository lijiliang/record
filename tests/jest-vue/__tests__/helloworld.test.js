import Vue from 'vue'
import { shallowMount } from '@vue/test-utils'
import HelloWorld from '@/components/HelloWorld.vue'

// describe('HelloWorld.vue', () => {
//   it('renders props.msg when passed', () => {
//     const root = document.createElement('div')
//     root.className = 'root'
//     document.body.appendChild(root)
//     new Vue({
//       render: h => h(HelloWorld, {
//         props: {
//           msg: 'Benson Li'
//         }
//       })
//     }).$mount('.root')
//     expect(document.getElementsByClassName('hello').length).toBe(1)
//     console.log(document.body.innerHTML)
//   })
// })


describe('HelloWorld.vue', () => {
  it('renders props.msg when passed', () => {
    const msg = 'new message'
    const wrapper = shallowMount(HelloWorld, {
      propsData: { msg }
    })
    console.log(wrapper.props('msg'))
    expect(wrapper.props('msg')).toEqual(msg)
    expect(wrapper.text()).toMatch(msg)
  })
})


describe(' HelloWorld.vue', () => {
  it('组件快照', () => {
    const wrapper = shallowMount(HelloWorld, {
      propsData: { msg : 'Benson Li'}
    })
    expect(wrapper).toMatchSnapshot();
  })
})
