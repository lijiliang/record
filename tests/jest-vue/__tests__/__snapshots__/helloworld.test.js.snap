// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` HelloWorld.vue 组件快照 1`] = `
<div class="hello">
  <h1>Benson Li</h1>
  test
</div>
`;
